Calico Networking K8s (#9)
Neokarm/neokarm-examples: k8s/calico-networking/README.md

# Calico Networking
This README refers to the internal Kubernetes network options that rke2 supports(CNIs)
## Choosing a CNI
rke2 server takes a parameter that refers to the cni to install.
The current supported options:
```
https://docs.rke2.io/install/network_options/
```
Default in this installation: ```calico``` will patch 
Default in rke2 would be ```canal```.

In order to change this theres a parameter in ```variables.tf``` called  ```cni```.

## Changing network encapsulation to bgp
- change ```bgp``` parameter in ```variables.tf``` to true.
A job will be created by rke to patch the requested change and this will attempt to rolling restart the daemonset of calico-node pods.
Monitor that all the calico-node pods get restarted, help if required.

> - Install calicoctl in order to interact with calico

Reviewer: add link to documentation

```
curl -L https://github.com/projectcalico/calico/releases/download/v3.22.0/calicoctl-linux-amd64 -o calicoctl
```
```
chmod +x ./calicoctl
```
This uses the applied ```KUBECONFIG``` env variable to connect.

- Apply ```bgpconfiguration.yaml``` to configure the nodes to work with the default ASnumber and node to node mesh.
```
calicoctl apply -f bgpconfiguration.yaml
```
Monitor that the calico-node pods get restarted correctly.

- Check if rke succeded in changing the IPPool in calico
```
calicoctl get ippool -oyaml
```
The IPPool should have
```
  ipipMode: Never
  natOutgoing: false
  nodeSelector: all()
  vxlanMode: Never
```
If this is not the case we will edit the IPPool with kubectl and correct the parameters
```
kubectl edit ippool <ippoolname>
```

> Again, Monitor the calico-node daemonset that should restart.

Reviewer: ```suggestion
> Important: Verify that all of the calico-node pods are restarting.
```
